scheduler/core/service/service.py

```diff
@@ -171,6 +171,12 @@ def _schedule_nights(night_indices: FrozenSet[NightIndex],
                             collector.time_accounting(plans,
                                                       sites=frozenset({site}),
                                                       end_timeslot_bounds=end_timeslot_bounds)
+                            if isinstance(update.event, MorningTwilightEvent):
```

Comment:
To avoid having to know about event types here, substitute:
```python
if update.done:
```

This should have the same ultimate effect without needing to muddy things with `isinstance`.